design/requirements/requirements.md

```diff
@@ -0,0 +1,22 @@
+Mandatory:
+
+- Test/quiz screen, Tests should be able to have their subjects selected(General OCA stuff, Inheritance, classes, etc.) and their length in questions.
+- User has to be able to login with username and password to see previous tests, results and so that previously failed questions can be asked again.
+- Logged in users can create new exam questions in a different screen. 
+    Questions should have the ability to easily add one or multiple blocks of code,
+    Should have the option to choose if it's a yes/no, multiple choice or multiple selection question type,
+    Should be declared which answer is correct and which are incorrect,
+    Should have the paragraph of the Gupta-book selected/shown,
+    Has to be given a tag that shows the subject.
+- Admin screen to oversee the tags, preferably implement checks to keep from duplicating tags.
+- Tags need to be able to be deleted if not attached to any questions
```

Comment:
if you wish to remove a tag that's attached to a question you should rename the tag (basic terms point 4)